core/templates/dev/head/domain/exploration/WrittenTranslationsObjectFactorySpec.ts

```diff
@@ -0,0 +1,195 @@
+        writtenTranslationsBackendDict.deleteContentId('content_2');
+      }).toThrow(Error('Unable to find the given content id.'));
+      expect(writtenTranslationsBackendDict.getAllContentId()).toEqual([
+        'content_1']);
+    });
+
+  it('should add translation in a written translations object', () => {
+    expect(() => {
+      writtenTranslationsBackendDict.addWrittenTranslation(
+        'content_1', 'hi-en', 'This is a HTML text');
+    }).toThrow(Error('Trying to add duplicate language code.'));
+
+    writtenTranslationsBackendDict.addWrittenTranslation(
+      'content_1', 'en', 'English HTML');
+    expect(writtenTranslationsBackendDict
+      .getTranslationsLanguageCodes('content_1')).toEqual(['hi-en', 'en']);
+  });
+
+  it('should update the html language code of a written translations object',
+    () => {
+      const writtenTranslationsBackendDict = (
+        writtenTranslationsObjectFactory.createFromBackendDict({
+          translations_mapping: {
+            content_1: {
+              'hi-en': {
+                html: '<p>This is the old HTML</p>',
+                needs_update: false
+              }
+            }
+          }
+        }));
+
+      expect(writtenTranslationsBackendDict.hasWrittenTranslation(
+        'content_1', 'hi-en')).toBe(true);
+      writtenTranslationsBackendDict.updateWrittenTranslationHtml(
+        'content_1', 'hi-en', '<p>This is the new HTML</p>');
+      expect(writtenTranslationsBackendDict.getWrittenTranslation(
+        'content_1', 'hi-en')).toEqual(
+        writtenTranslationObjectFactory.createFromBackendDict({
+          html: '<p>This is the new HTML</p>',
+          needs_update: false
+        }));
+
+      expect(() => {
+        writtenTranslationsBackendDict.updateWrittenTranslationHtml(
+          'content_1', 'en', 'This is the new HTML');
+      }).toThrow(Error('Unable to find the given language code.'));
+      expect(writtenTranslationsBackendDict.hasWrittenTranslation('en'))
+        .toBe(false);
+    });
+
+  it('should toggle needs_update for a language code', () => {
+    writtenTranslationsBackendDict.toggleNeedsUpdateAttribute(
+      'content_1', 'hi-en');
+    expect(writtenTranslationsBackendDict.getWrittenTranslation(
+      'content_1', 'hi-en')).toEqual(
+      writtenTranslationObjectFactory.createFromBackendDict({
+        html: '',
+        needs_update: true
+      }));
+    expect(writtenTranslationsBackendDict.hasUnflaggedWrittenTranslations(
+      'content_1')).toBe(false);
+
+    writtenTranslationsBackendDict.toggleNeedsUpdateAttribute(
+      'content_1', 'hi-en');
+    expect(writtenTranslationsBackendDict.getWrittenTranslation(
+      'content_1', 'hi-en')).toEqual(
+      writtenTranslationObjectFactory.createFromBackendDict({
+        html: '',
+        needs_update: false
+      }));
+    expect(writtenTranslationsBackendDict.hasUnflaggedWrittenTranslations(
+      'content_1')).toBe(true);
+  });
+
+  it('should blah', () => {
```

Comment:
Umm... was this supposed to be changed? :)

Reply:
Sorry about that!